pkg/cmd/plan.go

```diff
@@ -0,0 +1,212 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/ipfs/testground/pkg/api"
+	"github.com/ipfs/testground/pkg/config"
+
+	"github.com/BurntSushi/toml"
+	"github.com/go-git/go-git/v5"
+	gitcfg "github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/mattn/go-zglob"
+	"github.com/urfave/cli/v2"
+)
+
+var PlanCommand = cli.Command{
+	Name:  "plan",
+	Usage: "plan management",
+	Subcommands: cli.Commands{
+		&cli.Command{
+			Name:  "create",
+			Usage: "create plan `PLAN_NAME`",
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     "module",
+					Usage:    "Create module named `MODULE_NAME`",
+					Required: false,
+					Value:    "github.com/your/module/name",
+				},
```

Comment:
This command actually creates a **Golang** plan. It will fall short once we have SDKs for multiple languages (soon, with TypeScript next). Plans written in other languages will have to produce entirely different artifacts.

If we want to build this sustainably, we'll want to:

- introduce a flag: `--target=go`, `--target=ts-node`, `--target=ts-browser`.
- store all templates for a language under a directory with the target name.
- package those templates into the compiled `testground` binary via a tool like https://github.com/markbates/pkger.
- when this command is called, load the templates, invoke them, and produce the result.

Alternatively, how about storing these templates in a repo? `github.com/testground/templates`, with a directory per target env.

On creating a test plan, we could fetch the latest commit, pick the right subdirectory, and emit the templated artifacts?

Reply:
@coryschwartz ?

Reply:
I created a --target flag which currently just a simple map. I somehow missed the idea of using pkger. We could store these on another repo or download them on the fly as well.